### PR TITLE
Document generic synthetic bean creation

### PR DESCRIPTION
**Pull request by rmanibus**
Clarify how to create a generic synthetic bean in the `writing extensions` guide. 
other relevant places for this could be `cdi-integration` guide.

cc @mkouba

### CODE REVIEW

**Comment by mkouba** [comment]
I don't think it makes sense to add similar stuff to the `writing extensions` guide. The `cdi-integration` guide is the place where similar clarifications should go.

CC @Ladicek @manovotn

**Comment by rmanibus** [comment]
sounds good, I did move this in the cdi integration guide

**Comment by rmanibus** [comment]
not sure of the exact positioning in the doc, I feel that RN it is disrupting the progression between build time and runtime

File: docs/src/main/asciidoc/cdi-integration.adoc
Comment (Ladicek) [inline docs/src/main/asciidoc/cdi-integration.adoc:342]:
```suggestion
<1> `types()` must be used to specify the generic type.
```

**Comment by Ladicek** [comment]
Otherwise seems good.

File: docs/src/main/asciidoc/cdi-integration.adoc
Comment (Ladicek) [reply docs/src/main/asciidoc/cdi-integration.adoc:342]: Yeah, good point. (There's actually multiple overloads of `types()` and `addType()`, but that doesn't matter here.) It should be

> `types()` or `addType()` must be used to specify the generic type.

**Review by manovotn — APPROVED** [review]
Apart from the one pending comment, LGTM.
The position in the doc also fine :+1:

**Comment by gsmet** [comment]
@rmanibus I let you adjust the minor things? After that it's good to merge. Thanks!

**Comment by rmanibus** [comment]
@gsmet it's done !

**Comment by manovotn** [comment]
@Ladicek @mkouba can we merge this?

**Review by mkouba — APPROVED** [review]

**Pull request merged**
